src/client/src/apps/MainRoute/components/app-header/PWAInstallBanner/usePWABannerPrompt.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react'
+import { usePlatform } from 'rt-platforms'
+
+export interface InstallPromptEvent extends Event {
+  readonly platforms: string[]
+  readonly userChoice: Promise<{
+    outcome: 'accepted' | 'dismissed'
+    platform: string
+  }>
+  prompt(): Promise<void>
+}
+
+export const usePWABannerPrompt = (): [InstallPromptEvent | null, () => void] => {
+  const [prompt, setPrompt] = useState<InstallPromptEvent | null>(null)
+  const platform = usePlatform()
+
+  const promptToInstall = () => {
+    if (prompt) {
+      return prompt.prompt()
+    }
+    return Promise.reject(new Error("Browser hasn't sent a 'beforeinstallprompt' event"))
+  }
+
+  useEffect(() => {
+    const ready = (e: InstallPromptEvent) => {
+      e.preventDefault()
+      if (platform.type === 'browser') {
```

Comment:
I think I mentioned this during our chat, but this will also be true on mobile.

Reply:
On my testing the banner isn't displayed on mobile, as the `beforeinstallprompt` doesn't get fired. This just ensures it can't be triggered in a non-browser envrionment

Reply:
Are you testing on Chrome for Android?